### PR TITLE
Game

### PR DESCRIPTION
**Pull request by laurieTombola**
Huge pull request. Not sure how to do pull requests when you need everything in it to continue working.

### CODE REVIEW

**Comment by keithbarrow** [comment]
43 files. YOU ARE HAVENING\* A TURKISH
- Ref: https://www.chroniclesofgeorge.com/

**Comment by laurieTombola** [comment]
Yeah sorry, there was no point at which I could do a pull request without having to stop work till it was sorted.

File: main-app/app/scripts/game/controllers/game-controller.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
+(function () {
+    'use strict';
+    angular.module('Tombola.Games.NoughtsAndCrosses.Game')
+        .controller('Game', ['$scope', '$state', '$interval', 'GameServerProxy', 'PlayerType', 'GameStatus', function($scope, $state, $interval, proxy, playerType, game){
+            var me = this;
+            $scope.pageHeading = "Kittens Vs Puppies! Fighto!";
```
Comment (keithbarrow) [inline main-app/app/scripts/game/controllers/game-controller.js:6]: Just hard code in the HTML.

File: main-app/app/scripts/game/controllers/game-controller.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            $scope.turn = 1;
+            console.log('Game Controller Loaded');
```
Comment (keithbarrow) [inline main-app/app/scripts/game/controllers/game-controller.js]: remove

File: main-app/app/scripts/game/controllers/game-controller.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+            intialiseTurn();
+
+            $scope.board = function (){
+                return game.getBoard();
+            };
+
+            $scope.clickBox = function(box){
+                if(game.getBoard()[parseInt(box)] === '0' && game.getState() === 'Continue'){
+                    console.log('Box:  '+box+'||| Turn: '+$scope.turn);
```
Comment (keithbarrow) [inline main-app/app/scripts/game/controllers/game-controller.js]: Remove

File: main-app/app/scripts/game/controllers/game-controller.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+                                game.setBoard(setCharAt(game.getBoard(), box, $scope.turn));
+                                $scope.turn = $scope.turn === 1 ? 2 : 1;
+                                me.turnDelayPromise = $interval(function(){game.setBoard(response.data.gameboard);$scope.turn = $scope.turn === 1 ? 2 : 1;}, 2000, 1);
+                            }
+                            else{
+                                game.setBoard(response.data.gameboard);
+                            }
+                            game.setState(response.data.outcome);
+                            if(game.getState() === 'Draw'){
+                                console.log('Draw');
```
Comment (keithbarrow) [inline main-app/app/scripts/game/controllers/game-controller.js]: remove

File: main-app/app/scripts/game/controllers/game-controller.js [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+                            }
+                            else if(game.getState() === 'Win'){
+                                console.log('Win');
```
Comment (keithbarrow) [inline main-app/app/scripts/game/controllers/game-controller.js]: remove

File: main-app/app/scripts/game/controllers/game-controller.js (hunk 4 above)
Comment (keithbarrow) [inline main-app/app/scripts/game/controllers/game-controller.js]: remove

File: main-app/app/scripts/game/controllers/game-controller.js [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+                                $scope.pageHeading = 'Player ' + response.data.winner + 'Wins!';
+                                $interval(function(){
+                                    console.log('should be switching to win');
```
Comment (keithbarrow) [inline main-app/app/scripts/game/controllers/game-controller.js]: remove

File: main-app/app/scripts/game/controllers/game-controller.js [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+
+                            swapTurn(playerType.getPlayer1().type, playerType.getPlayer2().type);
+                        },
+                        function(response){
+                            // Failure
+                            console.log('Make Move Failed: '+ $scope.turn + ' |||||| ' + response.data);
+                        });
+                }
+                else {
+                    //console.log("Already Filled");
```
Comment (keithbarrow) [inline main-app/app/scripts/game/controllers/game-controller.js]: remove, plus the else condition

File: main-app/app/scripts/game/controllers/game-controller.js (hunk 1 above)
Comment (laurieTombola) [inline main-app/app/scripts/game/controllers/game-controller.js:6]: Its used later to notify players that a game is a win / draw

File: main-app/app/scripts/game/controllers/game-controller.js [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+                }
+            };
+
+            var swapTurn = function(player1Type, player2Type){
+                if(player1Type === 'human' && player2Type === 'human'){
+                    $scope.turn = $scope.turn === 1 ? 2 : 1;
+                }
+            };
+
+            var isHumanVsComputer = function(){
```
Comment (keithbarrow) [inline main-app/app/scripts/game/controllers/game-controller.js]: Whazz this into the playerType service the

File: main-app/app/scripts/game/controllers/noughts-and-crosses-controller.js [hunk 9]
```diff
@@ -0,0 +1,62 @@
+(function () {
+    'use strict';
+
+
+    angular.module('Tombola.Games.NoughtsAndCrosses')
```
Comment (keithbarrow) [inline main-app/app/scripts/game/controllers/noughts-and-crosses-controller.js]: Remove

File: main-app/app/scripts/game/services/make-move.js [hunk 10]
```diff
@@ -0,0 +1,23 @@
+(function () {
+    'use strict';
+
+    angular.module('Tombola.Games.NoughtsAndCrosses.Game')
```
Comment (keithbarrow) [inline main-app/app/scripts/game/services/make-move.js:4]: Make Move and new game is better in one service - correctly written the methods have 90% the same code.

File: main-app/app/scripts/lobby/controllers/lobby-controller.js [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+            $scope.player1 = function(){
+                return playerType.getPlayer1();
+            };
+
+            $scope.player2 = function(){
+                    return playerType.getPlayer2();
+                };
+
+            $scope.showTheGame = function(){
+                if(playerType.validatePlayerType(playerType.getPlayer1().type) && validatePlayerType(playerType.getPlayer2().type) ){
```
Comment (keithbarrow) [inline main-app/app/scripts/lobby/controllers/lobby-controller.js]: Put the validation in the service

File: main-app/app/scripts/navbar/controller/navbar-controller.js [hunk 12]
```diff
@@ -0,0 +1,11 @@
+(function () {
+    'use strict';
+
+    angular.module('Tombola.Games.NoughtsAndCrosses.Navbar')
```
Comment (keithbarrow) [inline main-app/app/scripts/navbar/controller/navbar-controller.js]: Does this do anything?

**Comment by keithbarrow** [comment]
:shipit: - But fix the problems highlighed in separate branches.

**Comment by laurieTombola** [comment]
Done.

**Pull request merged**